mmv1/third_party/terraform/services/sql/resource_sql_database_instance.go.erb

```diff
@@ -1880,6 +1880,33 @@ func resourceSqlDatabaseInstanceUpdate(d *schema.ResourceData, meta interface{})
 		}
 	}
 
+	// Check if edition changed and patch with datacache.
+	if d.HasChange("settings.0.edition") {
```

Comment:
One question about functionality: is it intended that you can only update "tier" if edition is upgraded? With how this code is setup, the code would not process if only "tier" was updated.

Reply:
Yes, this is intended; there are specific enterprise plus tiers that are only available if user specifies enterprise plus edition.

Reply:
This is implementing: https://cloud.google.com/sql/docs/mysql/upgrade-cloud-sql-instance-to-enterprise-plus-in-place 
For in place upgrade, edition (along with tier and data cache) must be changed on its own. I can add a comment in the code.

Reply:
@algawivem Would this lead to a potential state of a user trying to upgrade "tier" with the state apply being successful, but then receiving a permadiff as it wouldn't properly update?

Reply:
The argument "tier" is required so validate will fail if not provided.

If tier does not match the enterprise type, apply will fail for example: `Error: Error, failed to update instance settings for : googleapi: Error 400: Invalid request: Invalid Tier (db-perf-optimized-N-4) for (ENTERPRISE) Edition.`